### PR TITLE
Move flutter_gallery to the testing folder

### PR DESCRIPTION
**Pull request by mit-mit**
The main customer-focused Flutter Gallery demo app no longer lives in this repo, and is now being maintained out of the DevRel samples repo:
https://github.com/flutter/samples/tree/master/gallery

This PR moves flutter_gallery app -- which is still used for testing -- into the `dev/integration_tests/` folder.

### CODE REVIEW

**Comment by myappgurus** [comment]
Thanks

File: dev/devicelab/lib/tasks/web_dev_mode_tests.dart [hunk 1]
```diff
@@ -12,7 +12,7 @@ import '../framework/framework.dart';
 import '../framework/utils.dart';
 
 final Directory _editedFlutterGalleryDir = dir(path.join(Directory.systemTemp.path, 'edited_flutter_gallery'));
-final Directory flutterGalleryDir = dir(path.join(flutterDirectory.path, 'examples/flutter_gallery'));
+final Directory flutterGalleryDir = dir(path.join(flutterDirectory.path, 'dev/integration_tests/flutter_galleryutter_gallery'));
```
Comment (dnfield) [inline dev/devicelab/lib/tasks/web_dev_mode_tests.dart]: copy paste error?  `flutter_galleryutter_gallery`

**Review by dnfield — APPROVED** [review]
LGTM except for one typo

File: dev/devicelab/lib/tasks/web_dev_mode_tests.dart (hunk 1 above)
Comment (dnfield) [inline dev/devicelab/lib/tasks/web_dev_mode_tests.dart]:
```suggestion
final Directory flutterGalleryDir = dir(path.join(flutterDirectory.path, 'dev/integration_tests/flutter_gallery'));
```

File: dev/integration_tests/flutter_gallery/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Flutter gallery
+
+An older copy of the Flutter gallery demo application used for integration testing.
+
+For the current Flutter gallery app sample, see the samples repo:
+https://github.com/flutter/samples/tree/master/gallery
```
Comment (RedBrogdon) [inline dev/integration_tests/flutter_gallery/README.md]: The gallery just moved out of DevRel's repo and into https://github.com/flutter/gallery

There's also a hosted version at flutter.github.io/gallery.

Comment (mit-mit) [reply dev/integration_tests/flutter_gallery/README.md]: Links updated

File: dev/devicelab/lib/tasks/web_dev_mode_tests.dart (hunk 1 above)
Comment (mit-mit) [reply dev/devicelab/lib/tasks/web_dev_mode_tests.dart]: good catch, fixed

**Comment by mit-mit** [comment]
This caused two test failures; those were fixed in https://github.com/flutter/flutter/pull/52660

**Pull request merged**